cairosvg/parser.py

```diff
@@ -109,35 +116,29 @@ def __init__(self, node, parent=None, parent_children=False, url=None):
 
         # Inherits from parent properties
         if parent is not None:
-            items = parent.copy()
-            not_inherited = (
-                "transform", "opacity", "style", "viewBox", "stop-color",
-                "stop-opacity", "width", "height", "filter", "mask", "rotate",
-                "{http://www.w3.org/1999/xlink}href", "id", "x", "y",
-                "overflow", "clip", "clip-path")
-            for attribute in not_inherited:
-                if attribute in items:
-                    del items[attribute]
-
-            self.update(items)
+            self.update([
+                (a, parent[a]) for a in parent
+                if a not in __not_inherited_attrs
+            ])
             self.url = url or parent.url
             self.parent = parent
         else:
             self.url = getattr(self, "url", None)
             self.parent = getattr(self, "parent", None)
 
-        self.update(dict(self.node.attrib.items()))
+        self.update(self.node.attrib)
 
         # Give an id for nodes that don't have one
         if "id" not in self:
-            self["id"] = uuid.uuid4().hex
+            self["id"] = "node_%s_%x" %(id(self), random.getrandbits(32))
```

Comment:
`uuid.uuid4()` was surprisingly slow. This should provide Enough entropy to avoid any reasonable chance of collisions.